packages/react/src/__tests__/ReactStartTransition-test.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+let React;
+let ReactTestRenderer;
+let act;
+let useState;
+let useTransition;
+
+const SUSPICIOUS_NUMBER_OF_FIBERS_UPDATED = 10;
+
+describe('ReactStartTransition', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    ReactTestRenderer = require('react-test-renderer');
+    act = require('jest-react').act;
+    useState = React.useState;
+    useTransition = React.useTransition;
+  });
+
+  it('Warns if a suspicious number of fibers are updated inside startTransition', () => {
+    if (
+      require('shared/ReactFeatureFlags')
```

Comment:
In tests that only pass if a feature flag is enabled, you can use a `@gate` pragma:

```js
// @gate warnOnSubscriptionInsideStartTransition
it('name of test', () => {
  // write your test like you normally would
});
```

This will assert that the test is only expected to pass if the gate condition is true.

As a bonus, if it passes when the condition is false, we'll report it as a failure. This helps us tidy up old experiments.